.changesets/fix_geal_fix_hashing_algorithm.md

```diff
@@ -0,0 +1,26 @@
+### Fix cache key hashing algorithm ([Issue #5160](https://github.com/apollographql/router/issues/5160))
+
+> [!IMPORTANT]
+> If you have enabled [Distributed query plan caching](https://www.apollographql.com/docs/router/configuration/distributed-caching/#distributed-query-plan-caching), this release changes the hashing algorithm used for the cache keys.  On account of this, you should anticipate additional cache regeneration cost when updating between these versions while the new hashing algorithm comes into service.
+
+The Router contains a schema aware query hashing algorithm, designed to return the same hash across schema updates if they do not affect the query. It was used mainly to avoid planning a lot of queries when warming up the query plan cache. It was deactivated due to a regression. This reintroduces this algorithm, with a serie of reliability and performance fixes, along with better observability.
+
+The hashing algorithm can be activated in configuration, if query plan cache warm up is already enabled:
```

Comment:
I don't understand why cache warm up is mentioned here. There's no connection is there?

```suggestion
The hashing algorithm can be activated in configuration.
```

Reply:
it needs the warmup to be usable